crates/bevy_ui/src/render/mod.rs

```diff
@@ -447,37 +446,44 @@ pub fn extract_uinode_borders(
     default_ui_camera: Extract<DefaultUiCamera>,
     ui_scale: Extract<Res<UiScale>>,
     uinode_query: Extract<
-        Query<
-            (
-                &Node,
-                &GlobalTransform,
-                &ViewVisibility,
-                Option<&CalculatedClip>,
-                Option<&TargetCamera>,
-                Option<&Parent>,
-                &Style,
-                &BorderColor,
-            ),
-            Without<ContentSize>,
-        >,
+        Query<(
+            &Node,
+            &GlobalTransform,
+            &ViewVisibility,
+            Option<&CalculatedClip>,
+            Option<&TargetCamera>,
+            Option<&Parent>,
+            &Style,
+            AnyOf<(&BorderColor, &Outline)>,
+        )>,
     >,
     node_query: Extract<Query<&Node>>,
 ) {
     let image = AssetId::<Image>::default();
 
-    for (uinode, global_transform, view_visibility, clip, camera, parent, style, border_color) in
-        &uinode_query
+    for (
+        uinode,
+        global_transform,
+        view_visibility,
+        maybe_clip,
+        maybe_camera,
+        maybe_parent,
+        style,
+        (maybe_border_color, maybe_outline),
+    ) in &uinode_query
     {
-        let Some(camera_entity) = camera.map(TargetCamera::entity).or(default_ui_camera.get())
+        let Some(camera_entity) = maybe_camera
+            .map(TargetCamera::entity)
+            .or(default_ui_camera.get())
         else {
             continue;
         };
 
         // Skip invisible borders
         if !view_visibility.get()
-            || border_color.0.is_fully_transparent()
-            || uinode.size().x <= 0.
-            || uinode.size().y <= 0.
+            || style.display == Display::None
+            || maybe_border_color.is_some_and(|border_color| border_color.0.is_fully_transparent())
+                && maybe_outline.is_some_and(|outline| outline.color.is_fully_transparent())
```

Comment:
Could you move `maybe_border_color.is_some_and(|border_color| border_color.0.is_fully_transparent())
                && maybe_outline.is_some_and(|outline| outline.color.is_fully_transparent())` into an own variable for reading clarity?